docs(trailingSlashes): add note for SSG generation (#57628)
vercel/next.js: docs/02-app/02-api-reference/05-next-config-js/trailingSlash.mdx

With this option set, urls like `/about` will redirect to `/about/`.

> For SSG sites, this option can be used to generate `/about/index.html` files instead of the default `/about.html` files.

Reviewer: ```suggestion
When used with [`output: "export"`](https://nextjs.org/docs/app/building-your-application/deploying/static-exports) configuration, the `/about` page will output `/about/index.html` (instead of the default `/about.html`).
```
